Gems/Atom/RPI/Code/Source/RPI.Reflect/Material/MaterialAssetCreator.cpp

```diff
@@ -23,6 +23,7 @@ namespace AZ
             if (ValidateIsReady())
             {
                 m_asset->m_materialTypeAsset = parentMaterial.m_materialTypeAsset;
+                m_asset->m_materialTypeVersion = m_asset->m_materialTypeAsset->GetVersion();
```

Comment:
Required: This will only be correct if you apply the version updates to the MaterialSourceData before using it to create a MaterialAsset. Example...
1. The .materialtype has a property "twoSided". This is version 1.
2. The user creates a .material and sets the "twoSided" flag. The .material file is marked as material type version 1.
3. Someone updates the .materialtype to version 2, and renames "twoSided" to "doubleSided".
4. The material asset is loaded at runtime, sees the version mismatch, and applies the update thanks to your new runtime code. All is well.
5. Something causes the AP to reprocess the .material file.
6. The MaterialAssetCreator is given a property "twoSided". This doesn't exist in the property layout so there is some kind of error.

This is what needs to happen but is missing: 
- The MaterialBuilder runs the update steps to update the MaterialSourceData. This renames the .material file's "twoSided" to "doubleSided".
- The MaterialAssetCreator is given a value for "doubleSided" instead of "twoSided".
- The MaterialAsset is given the version number 2.
- No changes will be applied at runtime because they were applied at build time.